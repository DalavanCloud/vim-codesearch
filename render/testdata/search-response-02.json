{
    "search_response": [
        {
            "hit_max_results": false, 
            "status": 0, 
            "estimated_total_number_of_results": 2, 
            "search_result": [
                {
                    "docid": "A8sapeUr92w", 
                    "match_reason": {
                        "blame": false, 
                        "content": true, 
                        "filename_lineno": false, 
                        "scoped_symbol": true, 
                        "filename": false
                    }, 
                    "language": "c++", 
                    "best_matching_line_number": 23, 
                    "snippet": [
                        {
                            "scope": "base", 
                            "first_line_number": 20, 
                            "match_reason": {
                                "blame": false, 
                                "content": true, 
                                "filename_lineno": false, 
                                "scoped_symbol": true, 
                                "filename": false
                            }, 
                            "text": {
                                "text": "// version of the constructor, and if we are building a dynamic library we may\n// end up with multiple AtExitManagers on the same process.  We don't protect\n// this for thread-safe access, since it will only be modified in testing.\nstatic AtExitManager* g_top_manager = nullptr;\n\nstatic bool g_disable_managers = false;\n\n", 
                                "range": [
                                    {
                                        "range": {
                                            "start_line": 1, 
                                            "end_line": 2, 
                                            "start_column": 1, 
                                            "end_column": 1
                                        }, 
                                        "type": "SYNTAX_COMMENT"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 2, 
                                            "end_line": 3, 
                                            "start_column": 1, 
                                            "end_column": 1
                                        }, 
                                        "type": "SYNTAX_COMMENT"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 3, 
                                            "end_line": 4, 
                                            "start_column": 1, 
                                            "end_column": 1
                                        }, 
                                        "type": "SYNTAX_COMMENT"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 4, 
                                            "end_line": 4, 
                                            "start_column": 1, 
                                            "end_column": 7
                                        }, 
                                        "type": "SYNTAX_KEYWORD"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 4, 
                                            "end_line": 4, 
                                            "start_column": 23, 
                                            "end_column": 36
                                        }, 
                                        "type": "SNIPPET_QUERY_MATCH"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 4, 
                                            "end_line": 4, 
                                            "start_column": 39, 
                                            "end_column": 46
                                        }, 
                                        "type": "SYNTAX_KEYWORD"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 6, 
                                            "end_line": 6, 
                                            "start_column": 1, 
                                            "end_column": 7
                                        }, 
                                        "type": "SYNTAX_KEYWORD"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 6, 
                                            "end_line": 6, 
                                            "start_column": 8, 
                                            "end_column": 12
                                        }, 
                                        "type": "SYNTAX_KEYWORD"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 6, 
                                            "end_line": 6, 
                                            "start_column": 34, 
                                            "end_column": 39
                                        }, 
                                        "type": "SYNTAX_KEYWORD"
                                    }
                                ]
                            }
                        }, 
                        {
                            "scope": "base::AtExitManager::RegisterTask", 
                            "first_line_number": 62, 
                            "match_reason": {
                                "blame": false, 
                                "content": true, 
                                "filename_lineno": false, 
                                "scoped_symbol": false, 
                                "filename": false
                            }, 
                            "text": {
                                "text": "  AutoLock lock(g_top_manager->lock_);\n  DCHECK(!g_top_manager->processing_callbacks_);\n  g_top_manager->stack_.push(std::move(task));\n}\n\n// static\n", 
                                "range": [
                                    {
                                        "range": {
                                            "start_line": 1, 
                                            "end_line": 1, 
                                            "start_column": 17, 
                                            "end_column": 30
                                        }, 
                                        "type": "SNIPPET_QUERY_MATCH"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 2, 
                                            "end_line": 2, 
                                            "start_column": 11, 
                                            "end_column": 24
                                        }, 
                                        "type": "SNIPPET_QUERY_MATCH"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 3, 
                                            "end_line": 3, 
                                            "start_column": 3, 
                                            "end_column": 16
                                        }, 
                                        "type": "SNIPPET_QUERY_MATCH"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 6, 
                                            "end_line": 7, 
                                            "start_column": 1, 
                                            "end_column": 1
                                        }, 
                                        "type": "SYNTAX_COMMENT"
                                    }
                                ]
                            }
                        }, 
                        {
                            "scope": "base", 
                            "first_line_number": 95, 
                            "match_reason": {
                                "blame": false, 
                                "content": true, 
                                "filename_lineno": false, 
                                "scoped_symbol": false, 
                                "filename": false
                            }, 
                            "text": {
                                "text": "  DCHECK(g_top_manager->stack_.empty());\n}\n\nvoid AtExitManager::DisableAllAtExitManagers() {\n  AutoLock lock(g_top_manager->lock_);\n  g_disable_managers = true;\n}\n\nAtExitManager::AtExitManager(bool shadow)\n    : processing_callbacks_(false), next_manager_(g_top_manager) {\n  DCHECK(shadow || !g_top_manager);\n  g_top_manager = this;\n", 
                                "range": [
                                    {
                                        "range": {
                                            "start_line": 1, 
                                            "end_line": 1, 
                                            "start_column": 10, 
                                            "end_column": 23
                                        }, 
                                        "type": "SNIPPET_QUERY_MATCH"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 4, 
                                            "end_line": 4, 
                                            "start_column": 1, 
                                            "end_column": 5
                                        }, 
                                        "type": "SYNTAX_KEYWORD"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 5, 
                                            "end_line": 5, 
                                            "start_column": 17, 
                                            "end_column": 30
                                        }, 
                                        "type": "SNIPPET_QUERY_MATCH"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 6, 
                                            "end_line": 6, 
                                            "start_column": 24, 
                                            "end_column": 28
                                        }, 
                                        "type": "SYNTAX_KEYWORD"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 9, 
                                            "end_line": 9, 
                                            "start_column": 30, 
                                            "end_column": 34
                                        }, 
                                        "type": "SYNTAX_KEYWORD"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 10, 
                                            "end_line": 10, 
                                            "start_column": 29, 
                                            "end_column": 34
                                        }, 
                                        "type": "SYNTAX_KEYWORD"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 10, 
                                            "end_line": 10, 
                                            "start_column": 51, 
                                            "end_column": 64
                                        }, 
                                        "type": "SNIPPET_QUERY_MATCH"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 11, 
                                            "end_line": 11, 
                                            "start_column": 21, 
                                            "end_column": 34
                                        }, 
                                        "type": "SNIPPET_QUERY_MATCH"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 12, 
                                            "end_line": 12, 
                                            "start_column": 3, 
                                            "end_column": 16
                                        }, 
                                        "type": "SNIPPET_QUERY_MATCH"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 12, 
                                            "end_line": 12, 
                                            "start_column": 19, 
                                            "end_column": 23
                                        }, 
                                        "type": "SYNTAX_KEYWORD"
                                    }
                                ]
                            }
                        }
                    ], 
                    "num_duplicates": 0, 
                    "top_file": {
                        "display_name": {}, 
                        "file": {
                            "changelist": "52c2151398e310524ee0ceca4caad5e8d3857267", 
                            "name": "src/base/at_exit.cc", 
                            "package_name": "chromium"
                        }, 
                        "license": {}, 
                        "size": 0
                    }, 
                    "has_unshown_matches": true, 
                    "num_matches": 20, 
                    "is_augmented": false, 
                    "hit_max_matches": false
                }, 
                {
                    "docid": "svems5DwV8Q", 
                    "match_reason": {
                        "blame": false, 
                        "content": true, 
                        "filename_lineno": false, 
                        "scoped_symbol": false, 
                        "filename": false
                    }, 
                    "language": "c++", 
                    "best_matching_line_number": 112, 
                    "snippet": [
                        {
                            "scope": "kTSanDefaultSuppressions", 
                            "first_line_number": 109, 
                            "match_reason": {
                                "blame": false, 
                                "content": true, 
                                "filename_lineno": false, 
                                "scoped_symbol": false, 
                                "filename": false
                            }, 
                            "text": {
                                "text": "    \"race:base::Thread::StopSoon\\n\"\n\n    // http://crbug.com/272095\n    \"race:base::g_top_manager\\n\"\n\n    // http://crbug.com/308590\n    \"race:CustomThreadWatcher::~CustomThreadWatcher\\n\"\n", 
                                "range": [
                                    {
                                        "range": {
                                            "start_line": 1, 
                                            "end_line": 1, 
                                            "start_column": 5, 
                                            "end_column": 36
                                        }, 
                                        "type": "SYNTAX_STRING"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 3, 
                                            "end_line": 4, 
                                            "start_column": 5, 
                                            "end_column": 1
                                        }, 
                                        "type": "SYNTAX_COMMENT"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 4, 
                                            "end_line": 4, 
                                            "start_column": 5, 
                                            "end_column": 33
                                        }, 
                                        "type": "SYNTAX_STRING"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 4, 
                                            "end_line": 4, 
                                            "start_column": 17, 
                                            "end_column": 30
                                        }, 
                                        "type": "SNIPPET_QUERY_MATCH"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 6, 
                                            "end_line": 7, 
                                            "start_column": 5, 
                                            "end_column": 1
                                        }, 
                                        "type": "SYNTAX_COMMENT"
                                    }, 
                                    {
                                        "range": {
                                            "start_line": 7, 
                                            "end_line": 7, 
                                            "start_column": 5, 
                                            "end_column": 55
                                        }, 
                                        "type": "SYNTAX_STRING"
                                    }
                                ]
                            }
                        }
                    ], 
                    "num_duplicates": 0, 
                    "top_file": {
                        "display_name": {}, 
                        "file": {
                            "changelist": "9e475c6cd78c848504bef4b4939d1aca7619b24d", 
                            "name": "src/build/sanitizers/tsan_suppressions.cc", 
                            "package_name": "chromium"
                        }, 
                        "license": {}, 
                        "size": 0
                    }, 
                    "has_unshown_matches": false, 
                    "num_matches": 1, 
                    "is_augmented": false, 
                    "hit_max_matches": false
                }
            ], 
            "results_offset": 0, 
            "hit_max_matches_per_file": false, 
            "maybe_skipped_documents": false, 
            "hit_max_to_score": false
        }
    ]
}
