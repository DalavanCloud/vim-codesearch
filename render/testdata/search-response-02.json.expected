This file contains the rendered output per line and its associated metadata.

Lines that begin with a line number contains (after the '|') the contents that
will be inserted into the vim buffer at that line. The immediately following
line contains an object representing associated metadata.
-------------------------------------------------------------------------------

001|CodeSearch results for ^q{unspecified}q_
   |{}
002|
   |{}
003|1. src/base/at_exit.cc^>{
   |{'j': ('src/base/at_exit.cc', 0, 0)}
004|    20 ^c{// version of the constructor, and if we are building a dynamic library we may}c_
   |{'j': ('src/base/at_exit.cc', 19, 7)}
005|    21 ^c{// end up with multiple AtExitManagers on the same process.  We don't protect}c_
   |{'j': ('src/base/at_exit.cc', 20, 7)}
006|    22 ^c{// this for thread-safe access, since it will only be modified in testing.}c_
   |{'j': ('src/base/at_exit.cc', 21, 7)}
007|    23 ^k{static}k_ AtExitManager* ^${g_top_manager}$_ = ^k{nullptr}k_;
   |{'j': ('src/base/at_exit.cc', 22, 7)}
008|    24 
   |{'j': ('src/base/at_exit.cc', 23, 7)}
009|    25 ^k{static}k_ ^k{bool}k_ g_disable_managers = ^k{false}k_;
   |{'j': ('src/base/at_exit.cc', 24, 7)}
010|    26 
   |{'j': ('src/base/at_exit.cc', 25, 7)}
011|    27 }>_^>{
   |{'j': ('src/base/at_exit.cc', 26, 7)}
012|    [...]
   |{'j': ('src/base/at_exit.cc', 61, 9)}
013|    62   AutoLock lock(^${g_top_manager}$_->lock_);
   |{'j': ('src/base/at_exit.cc', 61, 7)}
014|    63   DCHECK(!^${g_top_manager}$_->processing_callbacks_);
   |{'j': ('src/base/at_exit.cc', 62, 7)}
015|    64   ^${g_top_manager}$_->stack_.push(std::move(task));
   |{'j': ('src/base/at_exit.cc', 63, 7)}
016|    65 }
   |{'j': ('src/base/at_exit.cc', 64, 7)}
017|    66 
   |{'j': ('src/base/at_exit.cc', 65, 7)}
018|    67 ^c{// static}c_
   |{'j': ('src/base/at_exit.cc', 66, 7)}
019|    68 }>_^>{
   |{'j': ('src/base/at_exit.cc', 67, 7)}
020|    [...]
   |{'j': ('src/base/at_exit.cc', 94, 9)}
021|     95   DCHECK(^${g_top_manager}$_->stack_.empty());
   |{'j': ('src/base/at_exit.cc', 94, 8)}
022|     96 }
   |{'j': ('src/base/at_exit.cc', 95, 8)}
023|     97 
   |{'j': ('src/base/at_exit.cc', 96, 8)}
024|     98 ^k{void}k_ AtExitManager::DisableAllAtExitManagers() {
   |{'j': ('src/base/at_exit.cc', 97, 8)}
025|     99   AutoLock lock(^${g_top_manager}$_->lock_);
   |{'j': ('src/base/at_exit.cc', 98, 8)}
026|    100   g_disable_managers = ^k{true}k_;
   |{'j': ('src/base/at_exit.cc', 99, 8)}
027|    101 }
   |{'j': ('src/base/at_exit.cc', 100, 8)}
028|    102 
   |{'j': ('src/base/at_exit.cc', 101, 8)}
029|    103 AtExitManager::AtExitManager(^k{bool}k_ shadow)
   |{'j': ('src/base/at_exit.cc', 102, 8)}
030|    104     : processing_callbacks_(^k{false}k_), next_manager_(^${g_top_manager}$_) {
   |{'j': ('src/base/at_exit.cc', 103, 8)}
031|    105   DCHECK(shadow || !^${g_top_manager}$_);
   |{'j': ('src/base/at_exit.cc', 104, 8)}
032|    106   ^${g_top_manager}$_ = ^k{this}k_;
   |{'j': ('src/base/at_exit.cc', 105, 8)}
033|    107 }>_
   |{'j': ('src/base/at_exit.cc', 106, 8)}
034|
   |{}
035|2. src/build/sanitizers/tsan_suppressions.cc^>{
   |{'j': ('src/build/sanitizers/tsan_suppressions.cc', 0, 0)}
036|    109     ^s{"race:base::Thread::StopSoon\n"}s_
   |{'j': ('src/build/sanitizers/tsan_suppressions.cc', 108, 8)}
037|    110 
   |{'j': ('src/build/sanitizers/tsan_suppressions.cc', 109, 8)}
038|    111     ^c{// http://crbug.com/272095}c_
   |{'j': ('src/build/sanitizers/tsan_suppressions.cc', 110, 8)}
039|    112     ^s{"race:base::^${g_top_manager}$_\n"}s_
   |{'j': ('src/build/sanitizers/tsan_suppressions.cc', 111, 8)}
040|    113 
   |{'j': ('src/build/sanitizers/tsan_suppressions.cc', 112, 8)}
041|    114     ^c{// http://crbug.com/308590}c_
   |{'j': ('src/build/sanitizers/tsan_suppressions.cc', 113, 8)}
042|    115     ^s{"race:CustomThreadWatcher::~CustomThreadWatcher\n"}s_
   |{'j': ('src/build/sanitizers/tsan_suppressions.cc', 114, 8)}
043|    116 }>_
   |{'j': ('src/build/sanitizers/tsan_suppressions.cc', 115, 8)}
044|
   |{}
045|
   |{}
